### PR TITLE
fix the ionIonBeam functional test

### PR DESCRIPTION
**Pull request by rochSmets**
This is a new version of the ionIonBeam test...

The former one used some unneeded latex command for producing the figure that bother the CI. They are now removed.

In the test, we now also calculate the real angular frequency of the mode with a decolvolution of the time series with the a `exp(gamma t)` function, the `gamma` being the interpolated value. But then the resulting expected cosine function is damped, with about 3 periods until `T=50`. We thus perform a `np.fft` to get the largest mode, but the result being too fragile, it does not appear a good idea to have an associated `assert`. The time evoluation of this cosine is displayed in the lower panel, so we can at least `visualy` check it on the CI

### CODE REVIEW

File: tests/functional/ionIonBeam/ion_ion_beam1d.py [hunk 1]
```diff
@@ -102,68 +102,91 @@ def vth(x):
...
+        ax2.text(0.40, 0.20, "omega (real) = {:5.3f}... expected 0.19".format(omega), transform=ax2.transAxes)
 
-        cov = np.fabs(np.array(pcov).flatten())
+        fig.savefig("ion_ion_beam1d.png")
 
-        assert np.fabs(popt[1]-0.09) < 1e-2
-        assert cov.max() < 1e-2
+        # compare with the values given gary et al. 1985
+        assert np.fabs(gamma-0.09) < 2e-2
+        #assert np.fabs(omega-0.19) < 8e-2
```
Comment (PhilipDeegan) [inline tests/functional/ionIonBeam/ion_ion_beam1d.py:189]: rm comment?

**Review by PhilipDeegan — COMMENTED** [review]
This test doesn't run on pull requests but I've verified it https://hephaistos.lpp.polytechnique.fr/teamcity/buildConfiguration/Phare_Phare_Users_Phil_Build/73668?buildTab=log&focusLine=4907&linesState=18.4955

**Review by PhilipDeegan — APPROVED** [review]

File: tests/functional/ionIonBeam/ion_ion_beam1d.py [hunk 2]
```diff
@@ -102,68 +102,91 @@ def vth(x):
...
+        by = by_interpolator(x)
+        bz = by_interpolator(x)
 
-        byz = np.append(byz, mm)
+        # get the mode 1, as it is the most unstable in a box of length 33
+        mode1 = np.absolute(np.fft.fft(by-1j*bz)[1])
+        first_mode = np.append(first_mode, mode1)
 
-    popt, pcov = curve_fit(yaebx, times, byz, p0=[0.1, 0.1]) # last args are initial guess
+    popt, pcov = curve_fit(yaebx, times, first_mode, p0=[0.08, 0.09])
```
Comment (nicolasaunai) [inline tests/functional/ionIonBeam/ion_ion_beam1d.py:134]: do we need to comment on those initial guesses which are quite close to the solution no ?

File: tests/functional/ionIonBeam/ion_ion_beam1d.py [hunk 3]
```diff
@@ -102,68 +102,91 @@ def vth(x):
...
+    damped_mode=first_mode*yaebx(times, 1/popt[0], -popt[1])
+
+    # find the omega for which "damped_mode" is the largest :
+    # this term is twice the one it should be because "mode1" resulting from
+    # an absolute value, this (cosx)^2 = cos(2x) then appears at the 2nd
+    # harmonoic (hence the factor 0.5 to get "omega")
+    # the factor "+1" is because we remove the DC component, so the value
+    # given by argmax has also to miss this value
+    omegas = np.fabs(np.fft.fft(damped_mode).real)
+    omega = 0.5*(omegas[1:omegas.size//2].argmax()+1)*2*np.pi/times[-1]
```
Comment (nicolasaunai) [inline tests/functional/ionIonBeam/ion_ion_beam1d.py:146]: there is "//2"

File: tests/functional/ionIonBeam/ion_ion_beam1d.py [hunk 4]
```diff
@@ -102,68 +102,91 @@ def vth(x):
...
+
+    print(omegas[0:6])
+    print(omegas[1:omegas.size//2].argmax())
```
Comment (nicolasaunai) [inline tests/functional/ionIonBeam/ion_ion_beam1d.py:149]: same //2

**Review by nicolasaunai — COMMENTED** [review]
few questions/comments

**Pull request merged**